pyiceberg/table/__init__.py

```diff
@@ -1627,6 +1632,67 @@ class AddFileTask:
     partition_field_value: Record
 
 
+class LocationProvider(ABC):
```

Comment:
I would also expect this one to be in `location.py`? The `table/__init__.py` is already pretty big